### PR TITLE
[angular] modal dialog for changing primary email

### PR DESCRIPTION
**Pull request by martinraison**
@andrewconner I also did some small refactoring in modal.js to remove repeated code. The idea is to have a kf-modal with general modal styling and logic, and a kf-modal-content inside of it (eg kf-basic-modal-content). This way we can reuse kf-modal when we create other modals in the future (for now I just modified the basic modal to add an optional Cancel button). Does that sound reasonable?

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/layout/footer/footer.tpl.html [hunk 1]
```diff
@@ -15,8 +15,10 @@ <h2 class="kf-footer-header">Getting Started with kifi</h2>
         <span class="kf-footer-item-icon kf-footer-item-icon-install"></span>
         Install the extension
       </a>
-      <div kf-basic-modal show="data.showInstall" title="Email verified" action-text="Continue">
-        Thanks for verifying your email address, George. You're good to go! Tell your friends, yo. Hahah, nah don't. You just verified your email. Thanks for verifying your email address, George. You're good to go! Tell your friends, yo. Hahah, nah don't. You just verified your email.
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/angular/src/layout/footer/footer.tpl.html:19]: (I assumed we don't want this in production)

**Comment by martinraison** [comment]
just noticed the email address doesn't show up in the modal - trying to fix it soon

File: kifi-backend/modules/shoebox/angular/src/layout/footer/footer.tpl.html (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/layout/footer/footer.tpl.html:19]: Hahaha. Yeah it was for reference only. It's not even in the correct location. Fine to remove.

**Comment by andrewconner** [comment]
Works for me, thanks. The repetition cleanup is good — I wanted two modal primitives (one for no custom styling, one for the "standard" styling), but this is a good compromise. One extra div to get the standard modal, in exchange for easier to understand code. :+1:

**Comment by andrewconner** [comment]
Feel free to merge when ready.

**Pull request merged**